met-api/src/met_api/resources/engagement_members.py

```diff
@@ -60,3 +60,20 @@ def post(engagement_id):
             return MembershipSchema().dump(member), HTTPStatus.OK
         except BusinessException as err:
             return {'message': err.error}, err.status_code
+
+
+@cors_preflight('GET,OPTIONS')
+@API.route('/<user_id>')
+class EngagementMembershipUser(Resource):
+    """Resource for fetching memberships for user."""
+
+    @staticmethod
+    @cross_origin(origins=allowedorigins())
+    @_jwt.has_one_of_roles([Role.VIEW_ASSIGNED_ENGAGEMENTS_SELF.value])
```

Comment:
who all will have this role (VIEW_ASSIGNED_ENGAGEMENTS_SELF)?

Reply:
Removed